src/shim.ts

```diff
@@ -6,43 +26,211 @@ import {
     Orchestrator,
 } from "./classes";
 import { DurableEntityBindingInfo } from "./durableentitybindinginfo";
-import { DurableOrchestrationBindingInfo } from "./durableorchestrationbindinginfo";
 import { OrchestratorState } from "./orchestratorstate";
+import { DurableOrchestrationInput } from "./testingUtils";
 
 /**
  * Enables a generator function to act as an orchestrator function.
  *
- * Orchestration context methods can be access
  * @param fn the generator function that should act as an orchestrator
- * @example Initialize an orchestrator
+ * @example Register an orchestrator using `app.generic()`
  * ```javascript
+ * const { app } = require("@azure/functions")
  * const df = require("durable-functions");
  *
- * module.exports = df.orchestrator(function*(context) {
- *     // function body
- * });
+ * app.generic('DurableFunctionsOrchestratorJS', {
+ *   trigger: trigger.generic({
+ *       type: 'orchestrationTrigger'
+ *   }),
+ *   handler: df.createOrchestrator(function* (context) {
+ *       // orchestrator body
+ *  })
+ * })
  * ```
  */
-export function orchestrator(
-    fn: (context: IOrchestrationFunctionContext) => Generator<unknown, unknown, any>
-): (
-    context: IOrchestrationFunctionContext,
-    orchestrationTrigger: DurableOrchestrationBindingInfo
-) => Promise<OrchestratorState> {
+export function createOrchestrator(fn: OrchestrationHandler): OrchestrationFunction {

```

Comment:
I guess we could combine them, but then we would be overloading the implementation, and I think that might be _more_ confusing, because they do subtly different but distinct things. `df.createOrchestrator()` takes in and `OrchestrationHandler` type (generator), and returns a `FunctionHandler` type -- something that can be passed to `app.generic()`. Meanwhile, `df.orchestration()` takes in an `OrchestrationHandler` generator and _registers_ it as a function on the function app, returning `void`. I don't see `createOrchestrator()` used much in production, but it could be useful for unit testing, for example see our own tests. Would be harder to have the same kind of tests with just `df.orchestration()`

Reply:
I don't want multiple overloads - I was suggesting we get rid of the "subtle but distinct things" in favor of one function that did one thing. It sounds like you still want "subtle but distinct" for unit tests? If so, I would challenge that a little bit. Both `createOrchestrator` and `orchestration` are only 2-3 lines each - I don't think the unit tests are really testing those functions in particular, but rather the stuff they're wrapping. Regardless, I wouldn't expose `createOrchestrator` or any relevant types outside of this package if its only needed for unit tests

Reply:
Well, I meant more for users' unit-testing, not necessarily us unit-testing. You're right, our unit tests aren't testing `df.createOrchestrator()` itself, but rather what it's wrapping, and same for users too -- it would help their unit tests

Reply:
Ah didn't realize you were trying to help with _users'_ unit testing. Are there existing guides/docs on how users should do unit testing in the old model? If so, please forward that to me and we can figure out a way to maintain the status quo. If not, unfortunately I think "users' unit testing" is going to be a way larger feature outside the scope of this PR and we should hide `createOrchestrator`/`OrchestrationFunction` because the names are too confusing.

Reply:
The testing story for DF in out-of-proc is unfortunately largely undocumented and community driving at this point. However, since this is a breaking change release, I'm not too worried about starting with a unit testing story from the get go.

At this time, `createOrchestrator` is a bit confusing for me as well. I would also like to punt `createOrchestrator` for this PR, and to discuss it's design in the context of unit testing in a follow-up PR.

Reply:
I'm happy with no longer exporting `createOrchestrator()` or revealing it as a public function, but I'd still like to keep it as a helper function, at least for this PR, because it's easier to keep our own existing unit tests as they are

Reply:
> Looks like you still need to remove OrchestrationFunction and EntityFunction from the public types.

Done.

> Btw, aren't these the same?

I'm not sure honestly. Idk why they were done the first way bc you're right the second one looks to be doing the same thing. @davidmrdavid do you know?

Reply:
I don't know, these decision predate me and. I prefer the simplified second version.

Reply:
Suffice to say I think there's still plenty of room for improvement for our own internal sake, but you removed the most confusing bits from being exported externally, so I will consider this resolved